### ISSUE TITLE
object detection run model_main.py all the evaluation result is -1.000

### ISSUE THREAD

**Issue by gfjiyue** [tensorflow/models#6370]
### System information
What is the top-level directory of the model you are using: Object detection
Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No, I just replace the sample data by my own.
OS Platform and Distribution: Ubuntu "16.04.2 LTS (Xenial Xerus)"
TensorFlow installed from (source or binary): installed with Python's pip package
TensorFlow version (use command below): ('v1.9.0-0-g25c197e023', '1.9.0')
CUDA/cuDNN version: CUDA 8.0
GPU model and memory: Tesla P40 22919MiB
### Exact command to reproduce:
python object_detection/model_main.py \
    --pipeline_config_path=${train_path}/data/faster_rcnn_resnet50_wheat.config \
    --model_dir=${train_path}/model \
    --num_train_steps=20000 \
    --sample_1_of_n_eval_examples=1 \
    --alsologtostderr

### Describe the problem
all the evaluation result is -1.000

### logs
2019-03-15 13:31:16.101496: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-03-15 13:31:16.433397: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1392] Found device 0 with properties: 
name: Tesla P40 major: 6 minor: 1 memoryClockRate(GHz): 1.531
pciBusID: 0000:03:00.0
totalMemory: 22.38GiB freeMemory: 22.22GiB
2019-03-15 13:31:16.433443: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1471] Adding visible gpu devices: 0
2019-03-15 13:31:16.765639: I tensorflow/core/common_runtime/gpu/gpu_device.cc:952] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-15 13:31:16.765689: I tensorflow/core/common_runtime/gpu/gpu_device.cc:958]      0 
2019-03-15 13:31:16.765699: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] 0:   N 
2019-03-15 13:31:16.766215: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1084] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 21555 MB memory) -> physical GPU (device: 0, name: Tesla P40, pci bus id: 0000:03:00.0, compute capability: 6.1)
/home/test/anaconda2/lib/python2.7/site-packages/matplotlib/__init__.py:1357: UserWarning:  This call to matplotlib.use() has no effect
because the backend has already been chosen;
matplotlib.use() must be called *before* pylab, matplotlib.pyplot,
or matplotlib.backends is imported for the first time.

  warnings.warn(_use_error_msg)
WARNING:tensorflow:Forced number of epochs for all eval validations to be 1.
W0315 13:31:17.710072 139895241332480 tf_logging.py:125] Forced number of epochs for all eval validations to be 1.
WARNING:tensorflow:Expected number of evaluation epochs is 1, but instead encountered `eval_on_train_input_config.num_epochs` = 0. Overwriting `num_epochs` to 1.
W0315 13:31:17.710439 139895241332480 tf_logging.py:125] Expected number of evaluation epochs is 1, but instead encountered `eval_on_train_input_config.num_epochs` = 0. Overwriting `num_epochs` to 1.
WARNING:tensorflow:Estimator's model_fn (<function model_fn at 0x7f3bb66b6b18>) includes params argument, but params are not passed to Estimator.
W0315 13:31:17.710735 139895241332480 tf_logging.py:125] Estimator's model_fn (<function model_fn at 0x7f3bb66b6b18>) includes params argument, but params are not passed to Estimator.
WARNING:tensorflow:num_readers has been reduced to 1 to match input file shards.
W0315 13:31:17.736921 139895241332480 tf_logging.py:125] num_readers has been reduced to 1 to match input file shards.
WARNING:tensorflow:From /home/test/tensorflow/models/research/object_detection/predictors/heads/box_head.py:93: calling reduce_mean (from tensorflow.python.ops.math_ops) with keep_dims is deprecated and will be removed in a future version.
Instructions for updating:
keep_dims is deprecated, use keepdims instead
W0315 13:31:23.238301 139895241332480 tf_logging.py:125] From /home/test/tensorflow/models/research/object_detection/predictors/heads/box_head.py:93: calling reduce_mean (from tensorflow.python.ops.math_ops) with keep_dims is deprecated and will be removed in a future version.
Instructions for updating:
keep_dims is deprecated, use keepdims instead
WARNING:tensorflow:From /home/test/tensorflow/models/research/object_detection/meta_architectures/faster_rcnn_meta_arch.py:2164: get_or_create_global_step (from tensorflow.contrib.framework.python.ops.variables) is deprecated and will be removed in a future version.
Instructions for updating:
Please switch to tf.train.get_or_create_global_step
W0315 13:31:23.305295 139895241332480 tf_logging.py:125] From /home/test/tensorflow/models/research/object_detection/meta_architectures/faster_rcnn_meta_arch.py:2164: get_or_create_global_step (from tensorflow.contrib.framework.python.ops.variables) is deprecated and will be removed in a future version.
Instructions for updating:
Please switch to tf.train.get_or_create_global_step
WARNING:tensorflow:From /home/test/tensorflow/models/research/object_detection/core/losses.py:345: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.
Instructions for updating:

Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.

See @{tf.nn.softmax_cross_entropy_with_logits_v2}.

W0315 13:31:24.927038 139895241332480 tf_logging.py:125] From /home/test/tensorflow/models/research/object_detection/core/losses.py:345: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.
Instructions for updating:

Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.

See @{tf.nn.softmax_cross_entropy_with_logits_v2}.

/home/test/anaconda2/lib/python2.7/site-packages/tensorflow/python/ops/gradients_impl.py:100: UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory.
  "Converting sparse IndexedSlices to a dense Tensor of unknown shape. "
2019-03-15 13:31:34.872445: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1471] Adding visible gpu devices: 0
2019-03-15 13:31:34.872534: I tensorflow/core/common_runtime/gpu/gpu_device.cc:952] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-15 13:31:34.872548: I tensorflow/core/common_runtime/gpu/gpu_device.cc:958]      0 
2019-03-15 13:31:34.872557: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] 0:   N 
2019-03-15 13:31:34.873210: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1084] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 21555 MB memory) -> physical GPU (device: 0, name: Tesla P40, pci bus id: 0000:03:00.0, compute capability: 6.1)
2019-03-15 13:41:43.156140: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1471] Adding visible gpu devices: 0
2019-03-15 13:41:43.156208: I tensorflow/core/common_runtime/gpu/gpu_device.cc:952] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-03-15 13:41:43.156219: I tensorflow/core/common_runtime/gpu/gpu_device.cc:958]      0 
2019-03-15 13:41:43.156227: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] 0:   N 
2019-03-15 13:41:43.156619: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1084] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 21555 MB memory) -> physical GPU (device: 0, name: Tesla P40, pci bus id: 0000:03:00.0, compute capability: 6.1)
creating index...
index created!
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=0.03s).
Accumulating evaluation results...
DONE (t=0.00s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = -1.000
WARNING:tensorflow:num_readers has been reduced to 1 to match input file shards.

**Comment by gfjiyue** [comment]
Maybe it was caused by the capital letter in class name. 
When I generate the tfrecord, there is an option 'classes_to_use=', I'm not sure if it is case sensitive or not.
So I rewrite all the class name into lowercase. 
Now it works well.
So please check your class name in annotation data carefully.

**Comment by jimwormold** [comment]
Thank you ever so much for this. Exactly the same issue, and solved by ensuring that the tf-record and the labe.pbtxt have the same case for the class. I had assumed that it used the class id, rather than the class text, but there we go!

**Comment by gfjiyue** [comment]
> Thank you ever so much for this. Exactly the same issue, and solved by ensuring that the tf-record and the labe.pbtxt have the same case for the class. I had assumed that it used the class id, rather than the class text, but there we go!
Glad to be helpful~
